### PR TITLE
Add clothing whitelist support

### PR DESCRIPTION
**Pull request by TiviPlus**
## About The Pull Request

Working on backend for robots

### CODE REVIEW

File: code/modules/mob/living/carbon/human/species.dm [hunk 1]
```diff
@@ -227,7 +230,10 @@
 		to_chat(prefs.parent, span_warning("You forgot to set your synthetic name in your preferences. Please do so next time."))
 
 /datum/species/proc/on_species_gain(mob/living/carbon/human/H, /datum/species/old_species)
-	return
+	for(var/slot_id in no_equip)
```
Comment (BraveMole) [inline code/modules/mob/living/carbon/human/species.dm:233]: SHOULD_CALL_PARENT ?

File: code/modules/mob/living/carbon/human/species.dm [hunk 2]
```diff
@@ -227,7 +230,10 @@
...
+		var/obj/item/thing = H.get_item_by_slot(slot_id)
+		if(thing && (!thing.species_exception || !is_type_in_list(src,thing.species_exception)))
```
Comment (BraveMole) [inline code/modules/mob/living/carbon/human/species.dm:235]: wouldn't !is_type_in_list(src, thing.species_exception) works on its own

File: code/game/objects/items.dm [hunk 3]
```diff
@@ -407,7 +414,8 @@
 		mob_equip = H.species.hud.equip_slots
 
 	if(H.species && !(slot in mob_equip))
-		return FALSE
+		if(!species_exception || !is_type_in_list(H.species, species_exception))
```
Comment (BraveMole) [inline code/game/objects/items.dm:417]: ditto

File: code/game/objects/items.dm [hunk 4]
```diff
@@ -57,6 +57,9 @@
 	var/slowdown = 0 // How much clothing is slowing you down. Negative values speeds you up
 	var/breakouttime = 0
 
+	///list() of species types, if a species cannot put items in a certain slot, but species type is in list, it will be able to wear that item
+	var/list/species_exception = null
```
Comment (BraveMole) [inline code/game/objects/items.dm:61]: i don't get why this is on the item, and not on the species

File: code/__HELPERS/string_lists.dm [hunk 5]
```diff
@@ -0,0 +1,14 @@
+GLOBAL_LIST_EMPTY(string_lists)
+
+/**
+ * Caches lists with non-numeric stringify-able values (text or typepath).
+ */
+/proc/string_list(list/values)
```
Comment (BraveMole) [inline code/__HELPERS/string_lists.dm:6]: can you explain to me why is this needed or better?

**Review by BraveMole — CHANGES_REQUESTED** [review]

File: code/__HELPERS/string_lists.dm (hunk 5 above)
Comment (TiviPlus) [reply code/__HELPERS/string_lists.dm:6]: it automatically caches the list to a string key

File: code/game/objects/items.dm (hunk 4 above)
Comment (TiviPlus) [reply code/game/objects/items.dm:61]: its cached

File: code/modules/mob/living/carbon/human/species.dm (hunk 1 above)
Comment (TiviPlus) [reply code/modules/mob/living/carbon/human/species.dm:233]: not required but done

File: code/modules/mob/living/carbon/human/species.dm (hunk 2 above)
Comment (TiviPlus) [reply code/modules/mob/living/carbon/human/species.dm:235]: done

File: code/game/objects/items.dm (hunk 3 above)
Comment (TiviPlus) [reply code/game/objects/items.dm:417]: done

**Review by BraveMole — APPROVED** [review]
